src/Data/Blob.hs

```diff
@@ -31,69 +31,52 @@ module Data.Blob
 
 import Prologue
 
+import           Analysis.File (fileBody)
+import qualified Analysis.File
+import           Analysis.Language as Language
 import           Control.Effect.Error
 import           Data.Aeson
 import qualified Data.ByteString.Lazy as BL
 import           Data.Edit
 import           Data.JSON.Fields
-import           Data.Language
 import           Data.Module
-import           Source.Source (Source)
+import           Source.Source (Source, totalSpan)
 import qualified Source.Source as Source
 import qualified System.FilePath as FP
 import qualified System.Path as Path
 import qualified System.Path.PartClass as Path.PartClass
 
--- | A 'FilePath' paired with its corresponding 'Language'.
--- Unpacked to have the same size overhead as (FilePath, Language).
-data File = File
-  { filePath     :: FilePath
-  , fileLanguage :: Language
-  } deriving (Show, Eq)
-
--- | Prefer 'fileForTypedPath' if at all possible.
-fileForPath :: FilePath  -> File
-fileForPath p = File p (languageForFilePath p)
-
-fileForTypedPath :: Path.PartClass.AbsRel ar => Path.File ar -> File
-fileForTypedPath = fileForPath . Path.toString
+type File = Analysis.File.File Language
 
 -- | The source, path information, and language of a file read from disk.
 data Blob = Blob
   { blobSource :: Source -- ^ The UTF-8 encoded source text of the blob.
   , blobFile   :: File   -- ^ Path/language information for this blob.
-  , blobOid    :: Text   -- ^ Git OID for this blob, mempty if blob is not from a git db.
```

Comment:
We no longer read items directly from Git, so the `oid` field was always empty. I’ve removed it.

Reply:
Nice 🤘